Source/Core/joinUrls.js

```diff
@@ -0,0 +1,84 @@
+/*global define*/
+define([
+        '../ThirdParty/Uri',
+        './defaultValue',
+        './defined',
+        './definedNotNull',
+        './DeveloperError'
+    ], function(
+        URI,
+        defaultValue,
+        defined,
+        definedNotNull,
+        DeveloperError) {
+    "use strict";
+
+    /**
+     * Function for joining URLs in a manner that is aware of query strings and fragments.
```

Comment:
I would make this `@private` or at least clean up this doc and mention this function in CHANGES.md.